### PR TITLE
[Feature][Connector-V2] Add influxDB connector source

### PR DESCRIPTION
**Pull request by 531651225**
https://github.com/apache/incubator-seatunnel/issues/1946
Add InfluxDB Connector Source

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by 531651225** [comment]
E2E
图片
图片

File: seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/source/InfluxDBSource.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    @Override
+    public void prepare(Config config) throws PrepareFailException {
+        CheckResult result = CheckConfigUtil.checkAllExists(config, URL, SQL);
+        if (!result.isSuccess()) {
+            throw new PrepareFailException(getPluginName(), PluginType.SOURCE, result.getMsg());
+        }
+        try {
+            this.influxDBConfig = new InfluxDBConfig(config);
+            InfluxDB influxDB = InfluxDBClient.getInfluxDB(influxDBConfig);
+            seaTunnelRowType = initTableField(influxDB, influxDBConfig);
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/source/InfluxDBSource.java]: hi, we need user config this. // 

        SeaTunnelSchema seatunnelSchema = SeaTunnelSchema.buildWithConfig(pluginConfig);
        this.typeInfo = seatunnelSchema.getSeaTunnelRowType();

https://github.com/apache/incubator-seatunnel/issues/2469

File: seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/source/InfluxDBSourceSplitEnumerator.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+                currentStart = currentStart - remainder;
+            }
+            query = sqls[0] + query;
+            if (sqls.length > 1) {
+                query = query + " and ( " + sqls[1] + " ) ";
+            }
+            influxDBSourceSplits.add(new InfluxDBSourceSplit(String.valueOf(i + System.nanoTime()), query));
+        }
+        return influxDBSourceSplits;
+    }
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/source/InfluxDBSourceSplitEnumerator.java]: extract this piece of logic as a public method? WDYT?

File: seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/client/InfluxDBClient.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+public class InfluxDBClient {
+    private static final long CONNECT_TIMEOUT = 15000;
+    public static InfluxDB getInfluxDB(InfluxDBConfig config) throws ConnectException {
+        OkHttpClient.Builder clientBuilder =
+                new OkHttpClient.Builder()
+                        .connectTimeout(CONNECT_TIMEOUT, TimeUnit.MILLISECONDS)
+                        .readTimeout(config.getQueryTimeOut(), TimeUnit.SECONDS);
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/client/InfluxDBClient.java]: Do we need to make this field configurable?

File: seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/source/InfluxDBSourceSplitEnumerator.java (hunk 2 above)
Comment (531651225) [reply seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/source/InfluxDBSourceSplitEnumerator.java]: > extract this piece of logic as a public method? WDYT?

It's a good suggestion. to abstract a public method to seatunnel-common module?

File: seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/resources/influxdb/influxdb_source_to_console.conf [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+transform {
+
+  # If you would like to get more information about how to configure seatunnel and see full list of transform plugins,
+  # please go to https://seatunnel.apache.org/docs/transform/sql
+}
+
+sink {
+  Console {}
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/resources/influxdb/influxdb_source_to_console.conf]: Use `Assert` sink check datatype?

File: seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/influxdb/InfluxDBSourceToConsoleIT.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+import java.io.IOException;
+import java.net.ConnectException;
+import java.sql.SQLException;
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+@Slf4j
+public class InfluxDBSourceToConsoleIT extends FlinkContainer {
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/influxdb/InfluxDBSourceToConsoleIT.java]: Can you add connector-v2 e2e-testcase(spark)？

File: seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/source/InfluxDBSource.java (hunk 1 above)
Comment (531651225) [reply seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/source/InfluxDBSource.java]: > hi, we need user config this. //
> 
> ```
>     SeaTunnelSchema seatunnelSchema = SeaTunnelSchema.buildWithConfig(pluginConfig);
>     this.typeInfo = seatunnelSchema.getSeaTunnelRowType();
> ```
> 
> #2469

hi，I had changed in new commit

File: seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/source/InfluxDBSourceSplitEnumerator.java (hunk 2 above)
Comment (531651225) [reply seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/source/InfluxDBSourceSplitEnumerator.java]: I had changed in new commit

File: seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/client/InfluxDBClient.java (hunk 3 above)
Comment (531651225) [reply seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/client/InfluxDBClient.java]: > Do we need to make this field configurable?

I had changed in new commit

File: seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/resources/influxdb/influxdb_source_to_console.conf (hunk 4 above)
Comment (531651225) [reply seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/resources/influxdb/influxdb_source_to_console.conf]: > Use `Assert` sink check datatype?

hi，I had changed e2e in new commit

File: seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/influxdb/InfluxDBSourceToConsoleIT.java (hunk 5 above)
Comment (531651225) [reply seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/influxdb/InfluxDBSourceToConsoleIT.java]: Spark E2E has a jar package conflict. Wechat groups in the community suggest paying attention to https://github.com/apache/incubator-seatunnel/issues/2607 . I wonder if I can add it after the community solves it.

File: docs/en/connector-v2/source/InfluxDB.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
+# InfluxDB
```
Comment (EricJoy2048) [inline docs/en/connector-v2/source/InfluxDB.md:1]: Hi, In https://github.com/apache/incubator-seatunnel/pull/2625 we  have redefined the format of the document. Please modify the document according to the current latest document format.

Comment (531651225) [reply docs/en/connector-v2/source/InfluxDB.md:1]: ok, I have modified it in new commit.

**Comment by 531651225** [comment]
@hailin0 @CalvinKirs @EricJoy2048  Hi brothers, please help me review. 
图片

 Integration test phase fails. How to solve this problem? My local E2E can pass

**Comment by hailin0** [comment]
image
https://pipelines.actions.githubusercontent.com/serviceHosts/5f1d2e93-1249-42e2-b16d-ed31c598c5d5/_apis/pipelines/1/runs/17499/signedlogcontent/36?urlExpires=2022-09-05T12%3A40%3A16.6214641Z&urlSigningMethod=HMACV1&urlSignature=pxCWdlIluLkqSV6lJ5ns5Fi9zSQKLy5Id%2FTDwE6xljY%3D



Please check for dependency jar conflicts

**Comment by 531651225** [comment]
> image
> https://pipelines.actions.githubusercontent.com/serviceHosts/5f1d2e93-1249-42e2-b16d-ed31c598c5d5/_apis/pipelines/1/runs/17499/signedlogcontent/36?urlExpires=2022-09-05T12%3A40%3A16.6214641Z&urlSigningMethod=HMACV1&urlSignature=pxCWdlIluLkqSV6lJ5ns5Fi9zSQKLy5Id%2FTDwE6xljY%3D
> 
> Please check for dependency jar conflicts
 Thank you for your pr https://github.com/apache/incubator-seatunnel/pull/2648. I solved the conflict for dependency jar  in 
flink e2e

File: docs/en/connector-v2/source/InfluxDB.md [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+### epoch [string]
+returned time precision
+- Optional values: H, m, s, MS, u, n
+- default value: n
+
+### query_timeout_sec [int]
+the `query_timeout` of the InfluxDB when you select, in seconds
+
+### connect_timeout_ms [long]
+the timeout for connecting to InfluxDB, in milliseconds
```
Comment (EricJoy2048) [inline docs/en/connector-v2/source/InfluxDB.md:134]: Please add some example for this connector. You can refer to the documentation of other connectors.

Comment (531651225) [reply docs/en/connector-v2/source/InfluxDB.md:134]: > Please add some example for this connector. You can refer to the documentation of other connectors.

 Thank you for your suggestion, I have added in new commit

**Review by EricJoy2048 — COMMENTED** [review]
LGTM. @hailin0  @CalvinKirs  PHAL

File: pom.xml [hunk 8]
```diff
@@ -433,6 +434,12 @@
                 <version>${jwt.version}</version>
                 <scope>runtime</scope>
             </dependency>
+
+            <dependency>
+                <groupId>org.influxdb</groupId>
+                <artifactId>influxdb-java</artifactId>
+                <version>${influxdb.version}</version>
+            </dependency>
```
Comment (ic4y) [inline pom.xml]: Connector-specific dependencies are now managed under seatunnel-connectors-v2/connector-influxdb/pom.xml

Comment (531651225) [reply pom.xml]: > Connector-specific dependencies are now managed under seatunnel-connectors-v2/connector-influxdb/pom.xml

ok, i have fixed in new commit

**Pull request closed without merging**